aspnetcore/security/authentication/configure-jwt-bearer-authentication.md

```diff
@@ -0,0 +1,365 @@
+
+> Note
+> Any app-to-app security would also work. Certificate authentication can be used or in Azure a managed identity can be used. 
+
+## Handling access tokens
+
+When using access tokens in a client application, the access tokens need to be rotated, persisted and stored somewhere on the server. In a web application, cookies are used to secure the session. The cookies should not contain the access tokens as the size of the cookie would become too large.
+
+Duende.AccessTokenManagement.OpenIdConnect is a great Nuget package for handling and managing access tokens in the client application. 
+
+> Note
+> If deploying the production, the cache should work in a mutli-instance deployment and a persistent cache is normally required.
+
+Some secure token servers encrypt the access tokens. Access tokens do not require any format. When using OAuth introspection, a reference token is used instead of an access token. A client (UI) application should never open an access token as the access token is not intended for this. Only an API for which the access token was created for should open the access token. 
+
+* Do not open access tokens in a UI application
+* Do not send the ID token to the APIs
+* Access tokens can have any format
+* Access tokens can be encrypted
+* Access tokens expire and need to be rotated
+* Access tokens are persisted on a secure backend server
+
+## YARP
+
+YARP (Yet Another Reverse Proxy) is a great tool for handling http requests and forwarding the requests to other APIs. YARP can implement security logic for acquiring new access credentials. YARP is used a lot in the backend for frontend (BFF) security architecture.
+
+## Testing APIs
+
+Testing secure APIs can be implemented in different ways. Integration tests and containers with access tokens can be used to test the APIs. You can also create the access tokens using the dotnet user-jwts tool.
+
+[Manage JSON Web Tokens in development with dotnet user-jwts](xref:security/authentication/jwt)
+
+It is important not to create security problems in the API so that you can test the API. When delegated access tokens are used, it is more difficult to test as the tokens can only be created using a UI and an OpenID Connect flow. If using a test tool to create the delegated access tokens, security features must be disabled for the tests. It is important that the features are only disabled in the test environment.
+
+### Use Swagger, Postman and other API UI tools
+
+Swagger and Postman are great UI tools for testing APIs. For the tools to work, the API can produce an Open API document and this can be loaded into the client testing tool. A security flow to acquire a new access token can be added to the API Open API file. 
+
+> Note 
+> Do not deploy insecure security test flows to production.
+
+If implementing a Swagger UI for an API, you should normally not deploy the UI to production as the security must be weakened to allow this to work. 
```

Comment:
```suggestion
Swagger UI for an API should only be enabled in the development environment, and never in production. Security must be lowered to enable Swagger UI .
```

Or, weaker version, less secure version

```suggestion
When implementing a Swagger UI for an API, you should normally not deploy the UI to production as the security must

folks who limit and control  access to the production app know they can skip this mandate.

Reply:
fixed, sometimes the UI is required in production, with weakened security headers